src/web/components/ParticipantRequests/ParticipantApprovalForm.tsx

```diff
@@ -0,0 +1,109 @@
+import { ChangeEvent, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
+import { ParticipantRequestDTO } from '../../../api/routers/participantsRouter';
+import { SiteDTO } from '../../../api/services/adminServiceClient';
+import { ParticipantApprovalFormDetails } from '../../services/participant';
+import { useSiteList } from '../../services/site';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { Input } from '../Input/Input';
+import { TextInput } from '../Input/TextInput';
+import { SearchBarContainer, SearchBarInput, SearchBarResults } from '../Search/SearchBar';
+
+import './ParticipantApprovalForm.scss';
+
+type ParticipantApprovalFormProps = {
+  onApprove: (formData: ParticipantApprovalFormDetails) => Promise<void>;
+  participant: ParticipantRequestDTO;
+  participantTypes: ParticipantTypeDTO[];
+};
+function ParticipantApprovalForm({
+  onApprove,
+  participant,
+  participantTypes,
+}: ParticipantApprovalFormProps) {
+  const { sites } = useSiteList();
+  const [searchText] = useState(participant.name);
+  const [selectedSite, setSelectedSite] = useState<SiteDTO>();
+
+  const formMethods = useForm<ParticipantApprovalFormDetails>({
+    defaultValues: { name: participant.name, types: participant.types?.map((t) => t.id) },
+  });
+  const { register, handleSubmit, setValue } = formMethods;
+
+  const onSubmit = async (data: ParticipantApprovalFormDetails) => {
+    await onApprove(data);
+  };
+
+  const onSiteClick = (site: SiteDTO) => {
+    setValue('siteId', site.id);
+    setSelectedSite(site);
+  };
+
+  const onSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    // This feature still under development :)
+    // setSearchText(event.target.value);
+  };
+
+  const getSiteText = (site: SiteDTO) => `${site.name} (Site ID ${site.id})`;
+
+  return (
+    <div className='participant-approval-form'>
+      <div>
+        To approve a participant please search Site IDs to see if they are already signed up.
+      </div>
+      <ul>
+        <li>{participant.name}</li>
+      </ul>
+      <SearchBarContainer>
+        <Input inputName='participantSearch' label='Search Participant Name to find Site ID'>
+          <SearchBarInput
+            inputClassName='search-input'
+            fullBorder
+            value={!selectedSite ? searchText : getSiteText(selectedSite)}
+            onChange={onSearchInputChange}
+            onFocus={() => setSelectedSite(undefined)}
+          />
+        </Input>
+        {!selectedSite && (
+          <SearchBarResults className='site-search-results'>
+            {sites?.map((s) => (
+              <button type='button' className='text-button' onClick={() => onSiteClick(s)}>
```

Comment:
I'm guessing you'll do this in a future PR, but it would be good if the whole element was clickable rather than just the text.

Reply:
We'll end up with multiple buttons, so probably just the text - I'll wait for feedback on this one I think once it's in integ.